File: app/src/main/java/com/demo/nytimesnews/base/presentation/BaseAdapter.java
package com.demo.nytimesnews.base.presentation;

import android.support.v7.widget.RecyclerView;

import java.util.List;

/**
 * Created by Rashida on 4/8/19.
 */
public abstract class BaseAdapter<T> extends RecyclerView.Adapter {

    protected List<T> list;

    public void setData(List<T> list) {
        this.list = list;
    }

    @Override
    public int getItemCount() {
        if (list != null && !list.isEmpty())
            return list.size();
        else
            return 0;
    }
}

File: app/src/main/java/com/demo/nytimesnews/base/presentation/BaseAdapter.kt
package com.demo.nytimesnews.base.presentation

import android.support.v7.widget.RecyclerView

/**
 * Created by Rashida on 4/8/19.
 */
abstract class BaseAdapter<T, E : RecyclerView.ViewHolder?> : RecyclerView.Adapter<E>() {
    protected var list: List<T>? = null
    fun setData(list: List<T>?) {
        this.list = list
        notifyDataSetChanged()
    }

    override fun getItemCount(): Int {
        return list?.size ?: 0
    }
}